spec/features/posts_index_by_user_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe 'User Post Index', type: :feature do
+  let!(:user) { User.create!(name: 'Tom', bio: 'Teacher from Mexico.', photo_url: 'https://unsplash.com/photos/F_-0BxGuVvo') }
+
+  before(:each) do
+    # Create posts for the user
+    @post1 = user.posts.create(title: 'Post 1', text: 'This is the first post.')
+    visit user_posts_path(user)
+  end
+
+  it 'displays the username and number of posts' do
+    # puts @post1.comments_counter
+    puts @post1.likes_counter
+    expect(page).to have_content("All Posts by #{user.name}")
+    expect(page).to have_content("Number of posts: #{user.post_counter}")
+  end
+
+  it "displays the post's title and part of the post's body" do
+    expect(page).to have_content('Post 1')
+    expect(page).to have_content('This is the first post.')
+  end
+
+  it 'displays the number of likes for each post' do
+    expect(page).to have_content('No likes yet')
+  end
+
+  it 'displays the number of comments for each post' do
+    expect(page).to have_content('No Comments yet')
+  end
```

Comment:
- [ ] As mentioned by the previous reviewer, these two tests do not actually test for the number of likes or comments a post has. These would throw errors whenever a post has likes or comments. Kindly create comments and likes for the post you've created and have the tests for them.